### DISCUSSION TITLE
The typings in Hono are awesome

### DISCUSSION THREAD

**Discussion by callmetwan** [honojs/hono#3698] [General]
@yusukebe I just wanted to say thank you for all your hard work. I've been using Hono for 6+ months I have to say this is the most well typed library I've used, _ever_.

I've learned so much just perusing the source code of Hono. Take MergePath:

```TypeScript
//https://github.com/honojs/hono/blob/ebdcd246b3d7106f51de3fe2e76e94eb4c42e0a5/src/types.ts#L1652-L1666
export type MergePath<A extends string, B extends string> = B extends ''
  ? MergePath<A, '/'>
  : A extends ''
  ? B
  : A extends '/'
  ? B
  : A extends `${infer P}/`
  ? B extends `/${infer Q}`
    ? `${P}/${Q}`
    : `${P}/${B}`
  : B extends `/${infer Q}`
  ? Q extends ''
    ? A
    : `${A}/${Q}`
  : `${A}/${B}`

```

I've seen some hacky solutions for enforcing path conventions, this is super elegant by comparison.

I've been diving into the typings for `hc` and it has blown my mind a bit (I think it will take me a long while to understand how you achieved it)

Anyways, just wanted to say thank you and share what I appreciate your hard work!!!

**Comment by yusukebe** [comment]
@callmetwan 

Thank you very much for your kind words!
